cmd/registry/scoring/parser.go

```diff
@@ -258,59 +257,131 @@ func validateBooleanThresholds(thresholds []*rpc.BooleanThreshold) []error {
+			return "", "", fmt.Errorf("input pattern %q does not match with target pattern %q", ip, tp)
 		}
 
-		// Check if the individual entities match
-		if tp.Name.ApiID != "-" && tp.Name.ApiID != r.Name.ApiID {
-			return fmt.Errorf("api mismatch in resource %s and target pattern %v", resourceName.String(), targetPattern)
+		// Merge the patters together
+		mergedPatternName := patterns.VersionName{
+			Name: names.Version{
+				ProjectID: projectID,
+			},
+		}
+		mergedApi, err := findCommonPattern(tp.Name.ApiID, ip.Name.ApiID)
+		if err != nil {
+			return "", "", fmt.Errorf("cannot find common pattern between %q and %q", tp.String(), ip.String())
 		}
-		if tp.Name.VersionID != "-" && tp.Name.VersionID != r.Name.VersionID {
-			return fmt.Errorf("version mismatch in resource %s and target pattern %v", resourceName.String(), targetPattern)
+		mergedPatternName.Name.ApiID = mergedApi
+		mergedVersion, err := findCommonPattern(tp.Name.VersionID, ip.Name.VersionID)
+		if err != nil {
+			return "", "", fmt.Errorf("cannot find common pattern between %q and %q", tp.String(), ip.String())
 		}
+		mergedPatternName.Name.VersionID = mergedVersion
+
+		// Merge the filters together
+		mergedFilter := generateCommonFilter(targetPattern.GetFilter(), inputFilter)
+
+		return mergedPatternName.String(), mergedFilter, nil
 	case patterns.ApiName:
-		// Check if targetPattern and resourceName match in type
-		r, ok := resourceName.(patterns.ApiName)
+		// Check if targetPattern and inputPattern match in type
+		ip, ok := inputPatternName.(patterns.ApiName)
 		if !ok {
-			return fmt.Errorf("resource %s doesn't match target pattern %s", r, tp)
+			return "", "", fmt.Errorf("input pattern %q does not match with target pattern %q", ip, tp)
 		}
 
-		// Check if the individual entities match
-		if tp.Name.ApiID != "-" && tp.Name.ApiID != r.Name.ApiID {
-			return fmt.Errorf("api mismatch in resource %s and target pattern %v", resourceName.String(), targetPattern)
+		// Merge the patters together
+		mergedPatternName := patterns.ApiName{
+			Name: names.Api{
+				ProjectID: projectID,
+			},
+		}
+		mergedApi, err := findCommonPattern(tp.Name.ApiID, ip.Name.ApiID)
+		if err != nil {
+			return "", "", fmt.Errorf("cannot find common pattern between %q and %q", tp.String(), ip.String())
 		}
+		mergedPatternName.Name.ApiID = mergedApi
+
+		// Merge the filters together
+		mergedFilter := generateCommonFilter(targetPattern.GetFilter(), inputFilter)
+
+		return mergedPatternName.String(), mergedFilter, nil
 	default:
-		return fmt.Errorf("unsupported resource type %T", targetPatternName)
+		return "", "", fmt.Errorf("Unsupported pattern in either targetPattern %q or inputPattern %q", targetPatternName.String(), inputPatternName.String())
 	}
+}
+
+func generateCommonFilter(a, b string) string {
+	if a == b {
+		return a
+	}
+	if a != "" && b != "" {
+		return fmt.Sprintf("(%s) && (%s)", a, b)
+	}
+	if a != "" && b == "" {
```

Comment:
Nit: these seem to be four mutually exclusive conditions, and here b = "" is implicit because we know that the test on 364 is false and a != "". I would find this clearer as a series of else-ifs.